core/message-tags-3.3.md

```diff
@@ -179,6 +201,19 @@ A message sent by a client with the `+example-client-tag` client-only tag:
 
 ---
 
+A message with the same tags sent with both the normal and JSON encodings for comparison:
+
+    @tag=value;+client-tag=value\swith\sspaces PRIVMSG #channel :Message
+    @{"tag":"value","+client-tag":"value\swith\spaces"} PRIVMSG #channel :Message
```

Comment:
Sure, that could work, space inside a string is the only thing that strictly needs to be escaped. I just though it would be simpler to describe and implement to just reuse the existing tag escaping routine that will already be available.

Encode: `'@' + tag_escape(json_encode_strip_whitespace(input_data)) + ' ' + rest_of_line`
Decode: `json_decode(tag_unescape(split_between_at_and_space(line)))`

But yeah, you could probably replace `tag_escape` with `replace_all(' ', '\u0020')` and skip the `tag_unescape`, just means more description in the spec and implementing an extra encode step (even if it's a simple one).